### PR TITLE
Pythonnet repl and evaluation

### PR DESCRIPTION
**Pull request by medegor44**
This PR introduce implementation of repl approach and Evaluate method in pythonnet runner.

### CODE REVIEW

**Comment by medegor44** [comment]
[Implement REPL and Evaluate in PythonNetRunner](https://trello.com/c/F2t5dWnU/25-implement-repl-and-evaluate-in-pythonnetrunner)

File: Tests/EmbeddedScripts.Python.PythonNet.Tests/PythonNetRunnerTests.cs [hunk 1]
```diff
@@ -151,12 +151,31 @@ public class A
...
+        public async Task Evaluate()
+        {
+            var runner = new PythonNetRunner();
+            var actual = await runner.EvaluateAsync<int>("1 + 1");
+            
+            Assert.Equal(2, actual);
+        }
+        
+        [Fact]
+        public async Task Evaluate1()
```
Comment (confirmit-horizons) [inline Tests/EmbeddedScripts.Python.PythonNet.Tests/PythonNetRunnerTests.cs:172]: nit: bad test name

File: Tests/EmbeddedScripts.Python.PythonNet.Tests/PythonNetRunnerTests.cs [hunk 2]
```diff
@@ -151,12 +151,31 @@ public class A
...
         [Fact]
-        public async void AddStructAsGlobals_Succeed()
+        public async Task AddStructAsGlobals_Succeed()
         {
             var runner = new PythonNetRunner();
             await runner
                 .Register(new A(), "a")
                 .RunAsync("a.X += 1;");
         }
+
```
Comment (confirmit-horizons) [inline Tests/EmbeddedScripts.Python.PythonNet.Tests/PythonNetRunnerTests.cs:161]: Probably more tests can be added (similar to other engines for repl cases)

File: Src/EmbeddedScripts.Python.PythonNet/PythonNetRunner.cs [hunk 3]
```diff
@@ -4,33 +4,46 @@
 
 namespace EmbeddedScripts.Python.PythonNet
 {
-    public class PythonNetRunner : ICodeRunner
+    public class PythonNetRunner : ICodeRunner, IEvaluator
     {
-        private readonly Container _container = new();
-
+        private PyScope _scope;
```
Comment (confirmit-horizons) [inline Src/EmbeddedScripts.Python.PythonNet/PythonNetRunner.cs:9]: Is PyScope a disposable object?

Comment (medegor44) [reply Src/EmbeddedScripts.Python.PythonNet/PythonNetRunner.cs:9]: Yes, it is. I'll fix that.

File: Tests/EmbeddedScripts.Python.PythonNet.Tests/PythonNetRunnerTests.cs (hunk 2 above)
Comment (medegor44) [reply Tests/EmbeddedScripts.Python.PythonNet.Tests/PythonNetRunnerTests.cs:163]: Some of new tests are red at this moment

**Comment by medegor44** [comment]
This PR is closed, because of there is no good approach to make pythonnet work properly in multithreaded environment (check issue https://github.com/pythonnet/pythonnet/issues/1110)

**Pull request closed without merging**